rustls/src/lib.rs

```diff
@@ -1,41 +1,66 @@
 //! # Rustls - a modern TLS library - portable-rustls fork
 //!
-//! <!-- TODO(portable-rustls) CLEANUP & IMPROVE NOTE FOR THIS FORK -->
-//! IMPORTANT NOTICE: REGARDLESS OF UPSTREAM `rustls` PROJECT THIS FORK IS NOT CERTIFIED AND NOT PEER-REVIEWED - USE AT YOUR OWN RISK AS STATED FURTHER BELOW
+//! <!-- TODO(portable-rustls) IMPROVE SYNCHRONIZATION OF IMPORTANT INFO WITH BEGINNING OF README.md -->
+//! <!-- (as tracked in: https://github.com/brody4hire/portable-rustls/issues/31) -->
 //!
 //! <!-- TODO(portable-rustls) CLEANUP & IMPROVE NOTE FOR THIS FORK -->
+//! __IMPORTANT NOTICE:__ regardless of upstream __`rustls`__ project this fork is __NOT CERTIFIED__ and __NOT PEER-REVIEWED__ - USE AT YOUR OWN RISK
+//!
+//! ## RECOMMENDED USAGE
+//!
 //! RECOMMENDED USAGE OF THIS FORK:
-//! * USE DEPENDENCY LIKE THIS IN `Cargo.toml`: `rustls = { package = "portable-rustls", ... }`
-//! * NEED TO EXPLICITLY ENABLE ANY FEATURES AS NEEDED - NO FEATURES ARE ENABLED BY DEFAULT IN THIS FORK
-//! * IMPORT AS USUAL FROM `rustls`: `use rustls;` OR `use rustls::...`
 //!
-//! <!-- TODO CLEAN UP THIS DOCUMENTATION -->
-//! THIS FORK SUPPORTS using `Arc` from `portable-atomic-util` to support targets with no atomic ptr, with the following requirements:
-//! * USE Rust nightly toolchain
-//! * USE `--cfg portable_atomic_unstable_coerce_unsized` in RUSTFLAGS FOR `cargo build` (etc.)
-//! * USE `--cfg unstable_portable_atomic_arc` in RUSTFLAGS FOR `cargo build` (etc.)
-//! * NEED TO ADD MULTIPLE DEPENDENCIES WITH SPECIFIC FEATURES ENABLED WHEN BUILDING FOR A TARGET WITH NO ATOMIC PTR, AS DESCRIBED DIRECTLY BELOW
+//! Add dependency on this fork as follows in `Cargo.toml`:
+//!
+//! ```rust,ignore
+//! rustls = { package = "portable-rustls", features=[...], ... }
+//! ```
+//!
+//! Then import and use __`rustls`__ in the code as usual.
+//!
+//! (Unlike the original __`rustls`__, no features are enabled by default in this fork.)
+//!
+//! ### targets with no atomic ptr
+//!
+//! This fork supports using `Arc` from `portable-atomic-util` to support targets with no atomic ptr, with the following requirements:
+//!
+//! Must use Rust nightly toolchain.
+//!
+//! Must use the following cfg flags in `RUSTFLAGS` FOR `cargo build` (etc.):
+//! - `--cfg portable_atomic_unstable_coerce_unsized`
+//! - `--cfg unstable_portable_atomic_arc`
```

Comment:
I may defer some of these updates to reduce the scope of this PR. Motivation for these updates was to avoid issue with nested bullet list, which is now resolved by recent update in e20cfeef4f6e46565be96ba9dc20253725630bcd.